src/v/redpanda/admin_server.cc

```diff
@@ -1072,27 +1074,35 @@ void admin_server::register_partition_routes() {
     ss::httpd::partition_json::get_partitions.set(
       _server._routes, [this](std::unique_ptr<ss::httpd::request>) {
           using summary = ss::httpd::partition_json::partition_summary;
-          return _partition_manager
-            .map_reduce0(
-              [](cluster::partition_manager& pm) {
-                  std::vector<summary> partitions;
-                  partitions.reserve(pm.partitions().size());
-                  for (const auto& it : pm.partitions()) {
-                      summary p;
-                      p.ns = it.first.ns;
-                      p.topic = it.first.tp.topic;
-                      p.partition_id = it.first.tp.partition;
-                      p.core = ss::this_shard_id();
-                      partitions.push_back(std::move(p));
-                  }
-                  return partitions;
-              },
-              std::vector<summary>{},
-              [](std::vector<summary> acc, std::vector<summary> update) {
-                  acc.insert(acc.end(), update.begin(), update.end());
-                  return acc;
-              })
-            .then([](std::vector<summary> partitions) {
+          auto get_summaries = [](auto& partition_manager) {
```

Comment:
For reviewers, here is a place where "coproc" code is invoked even when the flag is off. Shouldn't be an issue since the `coproc::partition_manager` should always be empty in those cases anyway.